Bug/5/resolve earthdata service unavailable

sberndtJPL opened the pull request:
Resolves #5 

1. Validates that the file size stored locally is the same size as the file on the server; if not, download the server file. The file sizes could differ if there was a download error, if internet went out, if the download was cancelled, or if podaacs server is unavailable
2. Checks if the downloaded / skipped local file is the same size as the PODAAC service unavailable file described in #5; deletes if so

Ultimately, this should make downloading more robust and enable larger scale data analysis without manually reviewing and downloading missing/invalid data

Optionally we could remove 1 as a parameter and do it by default. This would make the application more robust but less efficient as we would do this check for every file download. Let me know your thoughts.

Commit:
Validate file size of download & of previously downloaded files

Commit:
Validate local file size is same as remote by default.

Commit:
Update docstring to describe the function singularly as opposed to plurarl

sberndtJPL commented:
Example

![image](https://github.com/katjensen/cygnss-wetlands/assets/141465687/38305ba0-c50d-4f42-819b-339c545ae76e)

katjensen commented:
I think this is great ! I would be in favor of it being the default but defer to what you prefer

Commit:
Adds new parameter to validate_download to provide error message indicating earthdata system may be down

Commit:
Check if localfilesize is less than sourcefilesize to verify not replacing valid localfile with invalid earthdata system down file

Commit:
Update docstrings and variable names to more accurately describe the system down notification

sberndtJPL commented:
Ope! I actually opted to make it default last night but didn't update the comment of the PR. 

I added a couple new commits to notify the user if the earth data system is down, and, if so, not replace a potentially valid local file with the system down file. I also updated the docstrings and variable names to indicate the invalid file we're looking for is specifically the system down notification.

Commit:
Update docstring to indicate system down notification

sberndtJPL commented:
I think it would be best for you to re-review this since I did make some minor changes @katjensen

Commit:
Log failed downloads

Commit:
Fix bug to allow printing all successfully downloaded and failed files; not just the last date in the list of downloads

katjensen commented on cygnss_wetlands/cygnss/download.py:
```diff
@@ -1,6 +1,7 @@
 import base64
 import datetime
 import json
+import os.path
```
Should this import be simply `import os` ? Looks like both `os.path.getsize()` and `os.remove()` are called

(note: i just tested and it works as-is! I didn't realize this pattern was possible !)

sberndtJPL replied:
:o it is slightly concerning to me that this code works while only importing os.path lol

katjensen reviewed (approved):
Downloading works for me with updated code! I am not able to test handling of download errors easily -- I think a good to-do when we start adding tests would be to test these steps well
.
I just  have the one question about the import-- but otherwise looks good !

sberndtJPL commented:
I'll remove the .path portion of the os import and re-test.

Agree that it would be good to setup tests for this.

If you would like to test at the moment, running the script for 05-02-2023 will result in several failed downloads

katjensen commented:
Ooo thanks!! Yes trying out that date worked for me

Commit:
Remove .path from os.path since multiple os calls

sberndtJPL commented:
Great! 

I removed .path from os.path and retested. Worked as expected. Merging!
